### DISCUSSION TITLE
IDE not recognizing `Auth::user()->can();` ?

### DISCUSSION THREAD

**Discussion by richoandika** [spatie/laravel-permission#1808] [Q&A] [answered]
Can we calling using `Auth::user()->can();` ?
its works, but i have a little problem on IDE `Undefined method 'can'.`

_Originally posted by @richoandika in https://github.com/spatie/laravel-permission/issues/1287#issuecomment-894073702_

**Comment by richoandika** [comment]
I found a little idea to get rid of the error.

I added a new function in the User model:
![carbon (1)](https://user-images.githubusercontent.com/24300318/128490031-9d41bd3f-77fc-46fb-b78d-60d6de9aa120.png)
So i can use `(new User)->mySelf()->can();` on Controllers and that's to cover the mistakes that exist.

**Reply by fftfaisal** [reply]
Auth::user() already return the currently  logged in user model when you again search user by that user??

**Comment by erikn69** [comment]
`User::find` executes a sql
> on my text editor

Try
```php
/** @var \App\Models\User $user */
$user = Auth::user();
$user->can('dfdf')
```

**Reply by Shazzad998** [reply]
this still gives the same warning.

**Reply by erikn69** [reply, marked as answer]
```php
/** @var \App\Models\User $user */
```
weird, that comment must have told your IDE the class returned by the method

**Reply by ceciiiron** [reply]
Encountered the same problem with Laravel 10 and Laravel-spatie v5. This solution is better than the accepted answer because it does not execute a query
